### ISSUE TITLE
Playing some types of mp3 files can mute all other sounds that are being played (?)

### ISSUE THREAD

**Issue by mikeesouth** [ryanheise/just_audio#397]
**Which API doesn't behave as documented, and how does it misbehave?**
I guess it's AudioPlayer.play() that isn't behaving as expected.

**Minimal reproduction project**
Reproduced with the official example: https://github.com/mikeesouth/just_audio

**To Reproduce (i.e. user steps, not code)**
1. Tap "Stereo Song" below default controls in the example app
2. While the song is playing, tap "Test sound" and the sound should play "on top" of the song.
3. While the song is playing, tap "Testing 1 2 3" and notice how the song is muted (not paused) and the song resumes after "testing123.mp3" is played.

**Error messages**

```
No errors reported.
```

**Expected behavior**
I expect both "Test sound" and "Testing 1 2 3" to play on top of the song, i.e. the song should not be muted.
**NOTE!** This occurs on the Android emulator _(tested on Windows)_ **and on physical iOS devices** (tested on iPad mini 2nd generation) but not on my Android phone (physical device). On my Android phone, the testing 1 2 3 sound is played correctly "on top" of the song, i.e. it does not mute the song in the background.

**Smartphone (please complete the following information):**
 - Device: Android Emulator with Android 10 => The song is muted for the entire period when Testing 1 2 3 is played, i.e. JustAudio does **not** behave as expected.
 - Device: Physical iOS device (iPad Mini 2nd generation)  => The song is not muted entirely when Testing 1 2 3 is played but the song is muted or interrupted for a brief moment while Testing 1 2 3 is played but then the song is returned to "full volume" and the remaining part of "Testing 1 2 3" is played correctly on top of the song. The "Test sound" works perfectly, no interruption/mute of the background song. JustAudio does **not** behave as expected.
 - Device: Physical Android phone (Huawei Mate 20 Pro)  => The song is **not** muted when Testing 1 2 3 is played, i.e. JustAudio behaves as expected.

**Flutter SDK version**
```
[√] Flutter (Channel stable, 2.0.6, on Microsoft Windows [Version 10.0.19043.964], locale sv-SE)
[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[X] Chrome - develop for the web (Cannot find Chrome executable at C:\Program Files (x86)\Google\Chrome\Application)
    ! C:\Program Files (x86)\Google\Chrome\Application is not executable.
[!] Visual Studio - develop for Windows (Visual Studio Community 2019 16.9.0)
    X Visual Studio is missing necessary components. Please re-run the Visual Studio installer for the "Desktop development with C++" workload, and include these components:
        MSVC v142 - VS 2019 C++ x64/x86 build tools
         - If there are multiple build tool versions available, install the latest
        C++ CMake tools for Windows
        Windows 10 SDK
[√] Android Studio (version 4.1.0)
[√] VS Code (version 1.56.0)
[√] Connected device (3 available)

! Doctor found issues in 2 categories.
```

**Additional context**
The mp3 file that is causing the problem in the emulator and on iOS devices (testing123.mp3) is generated from Google Cloud Text-to-speech. The testing123.mp3 file is mono and has a sample rate of 24 000 Hz according to Audacity. I've tried to change test_sound.mp3 to mono with 24 000 Hz but it still works fine in the emulator and on iOS devices.
It seems like some other property than mono or 24 000 Hz of testing123.mp3 cause this issue.

EDIT: One more thing. If I convert testing123.mp3 to stereo using Audacity - then it works as expected in the emulator but iOS devices still have a brief mute/pause/interruption of the song.

UPDATE: I've tested on physical iOS devices and unfortunately the testing123.mp3 interrupts other sounds that are played. It's not as bad as on the emulator where other sounds are completely muted for the duration of testing123.mp3 is played but it's a brief interruption when testing123.mp3 is started. I'm not sure what property of testing123.mp3 that causes this but I guess someone with more knowledge of mp3 files can assist with this.

**Comment by mikeesouth** [comment]
Thanks for the reply and suggestion @ryanheise but unfortunately that's not a valid option for me. I'm playing these sound files from Google Text-to-speech on demand and frequently in the application. To handle the audio sessions manually is not a viable option, at least if I understand you correctly. I am using audioplayers library today and that library handles this without issues. However, audioplayers have alot of other issues that's handled much better in just_audio so I'm stuck between a rock and a hard place, kinda.

Oh well, I'm hoping that someone can find reason as to why this mp3 file behaves strangely on emulator and iOS so that we can circumvent that by either a) modify the file before playback or b) adjust the Google TTS call to get a file with other properties.

**Comment by ryanheise** [comment]
I'm not sure if I understand, what about Google TTS on demand prevents you from disabling the automatic audio session management of each player and letting your app handle it manually?

**Comment by mikeesouth** [comment]
This behavior has changed to the better but still not great when testing the latest version of just_audio (currently 0.8.0). Some type of mp3 files still cause a short interruption of currently played sounds on iOS, but it does no longer mute the currently played sounds on the Android emulator.
It's some property of the mp3 files that cause this behavior and it does not matter if I play the mp3 files from assets, file path or as buffered audio stream - as soon as play() is called it feels like the device "freezes" for some type of decoding before the sound is played and it also freezes briefly when the sound is completed and the resource is disposed(). The mp3 files that I've tested are less than 100 kb per file. I've tested on a pretty old iPad with iOS 12 so maybe this is a non issue for modern devices.

This is no longer a critical issue for me so I will go ahead and close this issue.
